MSAL/test/integration/native_auth/end_to_end/otp_code_retriever/MSALNativeAuthEmailCodeRetriever.swift

```diff
@@ -76,7 +83,19 @@ class MSALNativeAuthEmailCodeRetriever: XCTestCase {
             }
             if dataDictionary.count > 0 {
                 dataDictionary.sort(by: {($0["id"] as? Int ?? 0) > ($1["id"] as? Int ?? 0)})
-                return dataDictionary.first?["id"] as? Int
+                if let emailDateString = dataDictionary.first?["date"] as? String,
+                   let emailDate = MSALNativeAuthEmailCodeRetriever.dateFormatter.date(from: emailDateString) {
+                    let currentDate = Date()
+                    // Email should be newer than 5 seconds otherwise it could be from previous test
+                    // This retry will help with the delay in receiving the emails
+                    if currentDate.timeIntervalSince1970 - emailDate.timeIntervalSince1970  < 5 {
```

Comment:
Have you considered waiting 5 seconds and then check again? I'm saying this because here what we are doing is:

`[Date from CI computer] - [Date from email] has to be > 5 seconds`

What if there's a mismatch in the date from the email provider and the date in the CI (the region, etc.)? I mean, if the date from the email provider suddenly changes, we have no control over that.

Note that the downsize of this is that it will make the test execution slower.

What are your thoughts too @nilo-ms ?

Reply:
Technically there is a 4 second delay at the start of each retry:
`try? await Task.sleep(nanoseconds: UInt64(secondsToWait * Double(NSEC_PER_SEC)))`
But that doesn't seem to help enough

The issue I'm trying to solve here, is that on the first call it might be that the API didn't receive the email. This is strictly for SSPR to prevent using the previous email. I could technically wait 5 seconds and then get the email but that might not work also as the 4 second delay apparently doesn't work.
The dates are converted to UTC, regardless of the time `dateFormatter.timeZone = TimeZone(secondsFromGMT: 0) //We ignore the timezone` of the API. Currently the API is one hour before UTC and the code works so I expect if the timezone of the API changes, it will still work.

What needs to happen here, is for the `/challenge` endpoint to get called, try to get an email from the API for which the API specifies a receive time newer than 5 seconds (sometimes the API time is actually in the future, unclear of the reason. As in call to API is made at 15:40:00 and the API says email "came" at 15:40:02). If the email is not found, try again after a 4 second delay (`secondsToWait`) two more times, based on `numberOfRetry`

If I try to call the `/challenge` endpoint again, it will again go trough the same cycle.